Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/reissuer/DccReissuer.kt

```diff
@@ -25,30 +25,50 @@ class DccReissuer @Inject constructor(
         InvalidHealthCertificateException::class
     )
     suspend fun startReissuance(certificateReissuance: CertificateReissuance) {
-        certificateReissuance.migrateLegacyCertificate().certificates?.forEach {
+        val updates = certificateReissuance.asCertificateReissuanceCompat().certificates?.map {
             reissue(it)
         }
+        updates?.map {
+            it.recycleBin
+        }?.flatten()?.toSet()?.forEach {
+            moveToBin(it)
+        }
+
+        updates?.map {
+            it.register
+        }?.flatten()?.toSet()?.forEach {
+            register(it)
+        }
```

Comment:
`flatMap` ? with default emptySet of updates I think it can be enhanced a bit 

```Kotlin
       val updates =  ...}.orEmpty()

        updates.flatMap {
            it.recycleBin
        }.toSet().forEach {
            moveToBin(it)
        }
        
        updates.flatMap {
            it.register
        }.toSet().forEach {
            register(it)
        }
```